src/components/GroupBar.tsx

```diff
@@ -60,10 +60,17 @@ export default function GroupBar({
       if (!link) {
         setShowMeetingQuotaWarning(true);
       } else {
-        window.open(link, '_blank');
+        const newWindow = window.open(link, '_blank');
+        if (!newWindow || newWindow.closed || typeof newWindow.closed === 'undefined') {
+          // popup blocked
+          await launchMeetingInSameWindow(groupId);
```

Comment:
We shouldn't simply call launchMeetingInSameWindow() because calling the same RPC twice is wasteful and overall logic may not be correct. Merge these two functions instead. Note that their logic is similar but they have nuanced differences. Be very careful when change the code here. Make sure you fully understand the logic before making the change.